docs/building.md

```diff
@@ -199,3 +199,173 @@ added to the CMAKE_MODULE_PATH.
 ~~~~~~~~~~~~~
+
+## Building with Docker
+
+You can use `docker` and `docker-compose` to generate a complete build environment
+for mraa without having to install any other tool.
+
+Requirements:
+* [docker](https://www.docker.com/get-docker) >= 1.12.6
+* [docker-compose](https://docs.docker.com/compose/install/) >= 1.9.0
+
+**NOTE:** docker-compose is an optional requirement. It actually make running complex
+docker build and run command easier. But you can just use docker to build and run.
+
+### Docker Images Hierarchy
+
+To improve build times and images sizes, mraa use a build hierarchy to incrementally
+create a build environment. Find below a brief description of them:
+
+1. **mraa-base:** Provides the basic infrastructure and tools to compile C/C++ code and documentation.
+2. **mraa-python:** Provides the python2/python3 build tools. Depends on `mraa-base`.
+3. **mraa-java:** Provides the Java build tools. Depends on `mraa-base`.
+4. **mraa-android:** Provides the Android Things build tools.  Depends on `mraa-java`.
+5. **mraa-node4:** Provides the Node.js v4.4.7 build tools. Depends on `mraa-base`.
+6. **mraa-node5:** Same as `mraa-node4`, but using Node.js v5.12.0.
+
+**NOTE:** If you want to know which tools are installed for each of the mraa targets,
+just take a look at the `docker/` folder. All the related Dockerfiles are stored there!
+
+### Building Docker Images
+
+**tl;dr:** Just use this commands to build the hierarchy:
+
+```sh
+# Build the base image
+$ docker-compose build base
+# Build python image
+$ docker-compose build python
+# Build java image
+$ docker-compose build java
+# Build node4 image
+$ docker-compose build node4
+# Build node5 image
+$ docker-compose build node5
+# Build android things image
+$ docker-compose build android
+```
+
+**docker-compose** will take a look at the `docker-compose.yaml` file in the repository
+root directory, and build the requested target for you. At the end, docker-compose will
+tag the image built with an `mraa-` prefix. You can check them by running `docker images`.
+
+If you don't want to use docker-compose, you can also use `docker build` to generate every image.
+For example, to create the base image, you can do:
+
+```sh
+# From the repository root folder
+$ docker build -d docker/Dockerfile.base -t mraa-base .
+```
+
+Now, you don't actually need to build every image to start working. Let's say you
+are a Python developer, and has no idea what Node.js is, just build the base and
+python image!
+
+**NOTE:** If you work on Android Things, you will need the base, java, and android image.
+
+### Using Docker Images to build Mraa
+
+**tl;dr:** Just use this commands to build mraa:
+
+```sh
+# Build mraa documentation
+$ docker-compose run doc
+# Build mraa python2 package and run python2 tests
+$ docker-compose run python2
+# Build mraa python3 package and run python3 tests
+$ docker-compose run python3
+# Build mraa java package and run java tests
+$ docker-compose build java
```

Comment:
Fixed in https://github.com/intel-iot-devkit/mraa/pull/785/commits/6119f4478481d0941ed033e0d89db3f8b71cf06b